tachyon/base/profiler.h

```diff
@@ -0,0 +1,49 @@
+#ifndef TACHYON_BASE_PROFILER_H_
+#define TACHYON_BASE_PROFILER_H_
+
+#include <memory>
+#include <string>
+
+#include "third_party/perfetto/perfetto.h"
+
+#include "tachyon/base/files/file.h"
+
+PERFETTO_DEFINE_CATEGORIES(
+    perfetto::Category("Utils").SetDescription("Base utility functions"),
+    perfetto::Category("Subtask").SetDescription(
+        "Subtask within a bigger task"),
+    perfetto::Category("MSM").SetDescription(
+        "Multi Scalar Multiplication operations"),
+    perfetto::Category("ProofGeneration")
+        .SetDescription("The proof generation process"),
+    perfetto::Category("ProofVerification")
+        .SetDescription("The proof verification process"),
+    perfetto::Category("EvaluationDomain")
+        .SetDescription("Evaluation Domain operations"));
+
+namespace tachyon::base {
+
+class Profiler {
+ public:
+  Profiler() = default;
+  explicit Profiler(std::string_view trace_filepath, size_t max_size_kb);
+
+  void Init();
+  void DisableCategories(const std::string& category);
+  void EnableCategories(const std::string& category);
```

Comment:
```suggestion
  void DisableCategories(std::string_view category);
  void EnableCategories(std::string_view category);
```

Reply:
`add_disabled_categories` takes `std::string` so I believe `const std::string&` is better in this case.